src/diffusers/pipelines/wuerstchen/pipeline_wuerstchen_prior.py

```diff
@@ -0,0 +1,283 @@
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None,
+        latents: Optional[torch.FloatTensor] = None,
+        output_type: Optional[str] = "pt",
+        return_dict: bool = True,
+    ):
+        device = self._execution_device
+        do_classifier_free_guidance = guidance_scale > 1.0
+
+        if isinstance(num_inference_steps, int):
+            num_inference_steps = {0.0: num_inference_steps}
+
+        if isinstance(prompt, str):
+            prompt = [prompt]
+        elif not isinstance(prompt, list):
+            raise ValueError(f"`prompt` has to be of type `str` or `list` but is {type(prompt)}")
+
+        batch_size = len(prompt) if isinstance(prompt, list) else 1
+        text_encoder_hidden_states = self._encode_prompt(
+            prompt, device, num_images_per_prompt, do_classifier_free_guidance, negative_prompt
+        )
+
+        dtype = text_encoder_hidden_states.dtype
+        # latent_height = int(self.multiple * (height / self.multiple) / (1024 / 24))
+        latent_height = ceil(height / 42.67)
+        # latent_width = int(self.multiple * (width / self.multiple) / (1024 / 24))
+        latent_width = ceil(width / 42.67)
+        num_channels = self.prior.config.c_in
+        effnet_features_shape = (num_images_per_prompt * batch_size, num_channels, latent_height, latent_width)
+
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        latents = self.prepare_latents(effnet_features_shape, dtype, device, generator, latents)
+
+        for t in self.progress_bar(timesteps[:-1]):
+            ratio = t.expand(latents.size(0)).to(dtype)
+            predicted_image_embedding = self.prior(
+                torch.cat([latents] * 2) if do_classifier_free_guidance else latents,
+                r=torch.cat([ratio] * 2) if do_classifier_free_guidance else ratio,
+                c=text_encoder_hidden_states,
+            )
+
+            if do_classifier_free_guidance:
+                predicted_image_embedding_text, predicted_image_embedding_uncond = predicted_image_embedding.chunk(2)
+                predicted_image_embedding = torch.lerp(
+                    predicted_image_embedding_uncond, predicted_image_embedding_text, guidance_scale
+                )
+
+            latents = self.scheduler.step(
+                model_output=predicted_image_embedding,
+                timestep=ratio,
+                sample=latents,
+                generator=generator,
+            ).prev_sample
+
+        # normalize the latents
+        latents = latents * 42.0 - 1.0
+
+        if output_type == "np":
+            latents = latents.cpu().numpy()
+            text_encoder_hidden_states = text_encoder_hidden_states.cpu().numpy()
+
+        if not return_dict:
+            return (latents, text_encoder_hidden_states)
+
+        return WuerstchenPriorPipelineOutput(latents, text_encoder_hidden_states)
```

Comment:
why do we return the `text_encoder_hidden_states` here? They don't seem to be needed for the generator pipeline no?
```suggestion
        return WuerstchenPriorPipelineOutput(latents)
```